runtime/doc/eval.txt

```diff
@@ -3507,9 +3507,22 @@ getcurpos()	Get the position of the cursor.  This is like getpos('.'), but
 			MoveTheCursorAround
 			call setpos('.', save_cursor)
 <
-							*getcwd()*
-getcwd()	The result is a String, which is the name of the current
-		working directory.
+getcwd([{window}[, {tab}]])					*getcwd()*
+		With no argument the result is a String, which is the name of the
+		current effective working directory. If an arguments are provided for a
+		window or a tabpage the working directory of that scope is returned.
+
+		Tabs and windows are identified using their respective number. If a
+		number is 0 the current tab or window is implied. If one is not given
+		it defaults to 0. Thus
+		
+			getcwd()
+			getcwd(0)
+			getcwd(0, 0)
+
+		are all equal. If {window} is -1 it will be ignored and only the tab
+		will be looked at.
```

Comment:
While the task of checking for the exact tab local or window local working directory is more convoluted in this implementation than it would have been with scoping targets, the flexibility of targeting any window or tab and API compatibility with Vim probably are worth it, provided `haslocaldir()` semantics allow for sufficiently diffentiating working directory scopes. I find the docs a bit hazy on that topic, though: what does `haslocaldir(winnr, 0)` return when a tab local working directory is set for the tab containing `winnnr`, but not a window local one?